app/channel.go

```diff
@@ -806,6 +808,7 @@ func (a *App) UpdateChannelMemberNotifyProps(data map[string]string, channelId s
 		return nil, err
 	}
 
+	a.Srv.Store.Channel().InvalidateMembersForUser(userId)
```

Comment:
Is there any reason to not include `InvalidateMemberForUser` inside the `UpdateMember`, `RemoveMember` and `UpdateLastViewedAt` in the cache layer itself. That way you don't have to think about cache invalidation in the rest of the code, only from the store perspective.

Reply:
Exactly, I was thinking that too. But everywhere else, the invalidation calls happen from the app layer. Not sure why though. But yes, this can be done of course. Will update.

Reply:
Not everywhere else, some caches doesn't have any explicit invalidation to allow the application layer to invalidate them. They are self contained, for example the `role_layer.go` file. There all the cache and the invalidation is internal to the layer, and the app have no control of when this is invalidated or cached.

Reply:
Ahh that's great. Didn't see `role_layer.go`.